### ISSUE TITLE
Paginated API response returning duplicate racks

### ISSUE THREAD

**Issue by ebusto** [nautobot/nautobot#5991]
### Environment
* Nautobot version (Docker tag too if applicable): 1.6.26
* Python version: 3.8
* Database platform, version: PostgreSQL 14
* Middleware(s): N/A

### Steps to Reproduce
1. Create a large number of racks in Nautobot.
2. Paginate through the list of all racks.

### Expected Behavior
Each returned record to be distinct, by UUID.

### Observed Behavior
A small number of racks were duplicates.

This appears to be the exact issue described in this NetBox [issue](https://github.com/netbox-community/netbox/issues/3830), which was fixed in NetBox 2.7.0. Although this is before the version that was forked into Nautobot, this fix is not present in the latest 1.x and 2.x Nautobot branches.

**Comment by gsnider2195** [comment]
Is the order of these racks non-deterministic due to multiple racks with the same name? That was the issue that affected netbox but shouldn't apply here since we require a unique set of (`rack_group`, `name`) and order on (`location`, `rack_group`, `_name`)

**Comment by gsnider2195** [comment]
Actually, it looks like this is possible if `rack_group` is `null`. Do you have duplicate rack names with no rack group?

**Comment by ebusto** [comment]
Yes, we have a number of duplicate rack names with no rack group.

**Comment by timizuoebideri1** [comment]
Can be reproduced with
```python
import json, requests

headers = {'Authorization': 'Token fb6aefa227696d81cb9146a2f37c9004646a6c77', 'Content-type': 'application/json'}

for i in range(200):
    resp = requests.post('http://localhost:8082/api/dcim/racks/', data='{"name": "rack1", "status": "active", "site": "d6ded410-8aa5-4017-8682-586f69d6d38c"}', headers=headers)

url = 'http://localhost:8082/api/dcim/racks/?limit=10'
results = []

while url:
    data = json.loads(requests.get(url, headers=headers).content)
    url = data['next']
    results += data['results']

for index, result in enumerate(results):
    if results.count(result) > 1:
        print('duplicate', result['id'], 'at', index)

```
